.gitpod.yml

```diff
@@ -3,10 +3,18 @@ image:
 tasks:
   - name: tailscaled
     command: |
+      if [ -n "${TS_STATE_TAILSCALE_EXAMPLE}" ]; then
+        # restore the tailscale state from gitpod user's env vars
+        sudo mkdir -p /var/lib/tailscale
+        echo "${TS_STATE_TAILSCALE_EXAMPLE}" | sudo tee /var/lib/tailscale/tailscaled.state > /dev/null
+      fi
       sudo tailscaled
   - name: tailscale
     command: |
-      sudo -E tailscale up --hostname "gitpod-${GITPOD_WORKSPACE_ID}" \
-                           --authkey "${TAILSCALE_AUTHKEY}"
-
-experimentalNetwork: true
+      if [ -n "${TS_STATE_TAILSCALE_EXAMPLE}" ]; then
+        sudo -E tailscale up
+      else
+        sudo -E tailscale up --hostname "gitpod-${GITPOD_GIT_USER_NAME// /-}-$(echo ${GITPOD_WORKSPACE_CONTEXT} | jq -r .repository.name)"
+        # store the tailscale state into gitpod user
+        gp env TS_STATE_TAILSCALE_EXAMPLE="$(sudo cat /var/lib/tailscale/tailscaled.state)"
```

Comment:
instead of passing this though `gp env`, can't we place the state in `/workspace`? The actual content will be bound to the workspace lifecycle anyways, no?

Reply:
I want this state in all fresh workspaces for a certain project not only after restarting an existing one.